modules/apps/29-fee/keeper/keeper.go

```diff
@@ -85,3 +77,22 @@ func (k Keeper) ClaimCapability(ctx sdk.Context, cap *capabilitytypes.Capability
 	return k.scopedKeeper.ClaimCapability(ctx, cap, name)
 }
 */
+
+// SetCounterPartyAddress maps the counterparty relayer address to the source relayer address
+func (k Keeper) SetCounterpartyAddress(ctx sdk.Context, SourceAddress, CounterpartyAddress string) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeySourceAddress(SourceAddress), []byte(CounterpartyAddress))
+}
+
+// GetCounterPartyAddress gets the relayer counterparty address given a source address
+func (k Keeper) GetCounterPartyAddress(ctx sdk.Context, sourceAddress sdk.AccAddress) (sdk.AccAddress, bool) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.KeySourceAddress(sourceAddress.String())
+
+	if !store.Has(key) {
+		// TODO: add logging here
```

Comment:
why is logging useful here?

Reply:
Hmm, I was thinking of a `DEBUG` log for the rare case that you're trying to figure out why something isn't working. 

Hadn't put much thought into it though tbh. Patterns for logging are always a funny topic of conversation.

Reply:
I'm thinking there might be times when the key not existing makes sense. Might make more sense to log where this function is called from (if we are certain the key should exist)

Reply:
Happy to leave it out for now. :+1: